contracts/DCAFeeManager/DCAFeeManager.sol

```diff
@@ -1,33 +1,118 @@
+      if (_platformBalance > 0) {
+        IDCAHub.AmountOfToken[] memory _amountToWithdraw = new IDCAHub.AmountOfToken[](1);
+        _amountToWithdraw[0] = IDCAHub.AmountOfToken({token: address(wToken), amount: _platformBalance});
+        hub.withdrawFromPlatformBalance(_amountToWithdraw, address(this));
+      }
+    }
+
+    // Withdraw wToken from positions
+    if (_positionIds.length > 0) {
+      IDCAHub.PositionSet[] memory _positionSets = new IDCAHub.PositionSet[](1);
+      _positionSets[0] = IDCAHubPositionHandler.PositionSet({token: address(wToken), positionIds: _positionIds});
+      hub.withdrawSwappedMany(_positionSets, address(this));
+    }
+
+    // Unwrap and transfer
+    uint256 _totalBalance = wToken.balanceOf(address(this));
+    if (_totalBalance > 0) {
+      wToken.withdraw(_totalBalance);
+      _recipient.transfer(_totalBalance);
+    }
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function withdrawFromPlatformBalance(IDCAHub.AmountOfToken[] calldata _amountToWithdraw, address _recipient) external onlyOwnerOrAllowed {
+    hub.withdrawFromPlatformBalance(_amountToWithdraw, _recipient);
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function withdrawFromBalance(IDCAHub.AmountOfToken[] calldata _amountToWithdraw, address _recipient) external onlyOwnerOrAllowed {
+    for (uint256 i; i < _amountToWithdraw.length; i++) {
+      IERC20(_amountToWithdraw[i].token).safeTransfer(_recipient, _amountToWithdraw[i].amount);
+    }
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function withdrawFromPositions(IDCAHub.PositionSet[] calldata _positionSets, address _recipient) external onlyOwnerOrAllowed {
+    hub.withdrawSwappedMany(_positionSets, _recipient);
   }
 
   /// @inheritdoc IDCAFeeManager
-  function targetTokensDistribution() external view returns (TargetTokenShare[] memory) {
-    return _distribution;
+  function fillPositions(AmountToFill[] calldata _amounts, TargetTokenShare[] calldata _distribution) external onlyOwnerOrAllowed {
+    for (uint256 i; i < _amounts.length; i++) {
+      AmountToFill memory _amount = _amounts[i];
+
+      if (IERC20(_amount.token).allowance(address(this), address(hub)) == 0) {
+        // Approve the token so that the hub can take the funds
+        IERC20(_amount.token).approve(address(hub), type(uint256).max);
+      }
+
+      // Distribute to different tokens
+      uint256 _amountSpent;
+      for (uint256 j; j < _distribution.length; j++) {
+        uint256 _amountToDeposit = j < _distribution.length - 1
+          ? (_amount.amount * _distribution[j].shares) / MAX_TOKEN_TOTAL_SHARE
+          : _amount.amount - _amountSpent; // If this is the last token, then assign everything that hasn't been spent. We do this to prevent unspent tokens due to rounding errors
+
+        bool _failed = _depositToHub(_amount.token, _distribution[j].token, _amountToDeposit, _amount.amountOfSwaps);
+        if (!_failed) {
+          _amountSpent += _amountToDeposit;
+        }
```

Comment:
Should we revert here? Instead of try/catching. Try/catching and doing this would lead to unintended behavior. Obviously, its not big deal since its our fees... But, why should we keep going if something failed?